lib/model/message_list.dart

```diff
@@ -86,6 +87,30 @@ class MessageListView extends ChangeNotifier {
     notifyListeners();
   }
 
+  Message applyChangesToMessage(UpdateMessageEvent event, Message oldMessage) {
+    Map<String, dynamic> oldMessageJson = oldMessage.toJson();
```

Comment:
Per our discussion in `#mobile-team`, the idea here is to mutate the old message with the event data. However, it looks like `Message` and its implementations `DmMessage` and `StreamMessage` are immutable. This was my workaround, but it's a little clunky. Thoughts?

Reply:
Works for me! I'll do that instead of this json-switcheroo.